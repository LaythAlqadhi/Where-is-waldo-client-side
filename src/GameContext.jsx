import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

const initialState = {
  state: { loading: false, error: null },
  status: { isStarted: false, isEnded: true },
  message: null,
};

export const GameContext = createContext(initialState);

export function GameProvider({ children }) {
  const [state, setState] = useState({ loading: false, error: null });
  const [status, setStatus] = useState({
    isStarted: false,
    isEnded: false,
    message: null,
  });
  const [message, setMessage] = useState(null);

  return (
    <GameContext.Provider
      value={{
        state,
        setState,
        status,
        setStatus,
        message,
        setMessage,
      }}
    >
      {children}
    </GameContext.Provider>
  );
}

GameProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
