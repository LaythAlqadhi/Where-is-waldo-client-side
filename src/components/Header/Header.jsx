import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { GameContext } from '../../GameContext';

function Header({ timer }) {
  const { status, setStatus } = useContext(GameContext);

  return (
    <header>
      <Link
        to="/"
        onClick={() =>
          setStatus({
            isStarted: false,
            isEnded: false,
          })
        }
        aria-label="Logo"
      >
        Dragon Charmer&apos;s Island
      </Link>
      {timer && status.isStarted && timer}
    </header>
  );
}

Header.propTypes = {
  timer: PropTypes.element,
};

Header.defaultProps = {
  timer: null,
};

export default Header;
