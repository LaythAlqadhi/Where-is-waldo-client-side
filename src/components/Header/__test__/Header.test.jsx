import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';
import { GameContext } from '../../../GameContext';
import Header from '../Header';

it('renders Header component correctly', () => {
  const setStatus = vi.fn();

  const { asFragment } = render(
    <MemoryRouter>
      <GameContext.Provider
        value={{ status: { isStarted: true, isEnded: false }, setStatus }}
      >
        <Header timer={<div>00:00:00</div>} />
      </GameContext.Provider>
    </MemoryRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});
