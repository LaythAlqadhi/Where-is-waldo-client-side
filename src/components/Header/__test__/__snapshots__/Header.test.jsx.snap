// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Header component correctly 1`] = `
<DocumentFragment>
  <header>
    <a
      aria-label="Logo"
      href="/"
    >
      Dragon Charmer's Island
    </a>
    <div>
      00:00:00
    </div>
  </header>
</DocumentFragment>
`;
